rpc-rs/src/provider_main.rs

```
@@ -290,90 +295,164 @@ where
+            let layered = base_reg.with(level_filter).with(log_layer).with(tracing_layer);
+            tracing::subscriber::set_global_default(layered)
         }
-        match opentelemetry_otlp::new_pipeline()
-            .tracing()
-            .with_exporter(
-                opentelemetry_otlp::new_exporter()
-                    .http()
-                    .with_endpoint(tracing_endpoint)
-                    .with_protocol(Protocol::HttpBinary),
-            )
-            .with_trace_config(
-                trace::config()
-                    .with_sampler(Sampler::AlwaysOn)
-                    .with_id_generator(IdGenerator::default())
-                    .with_max_events_per_span(64)
-                    .with_max_attributes_per_span(16)
-                    .with_max_events_per_span(16)
-                    .with_resource(Resource::new(vec![opentelemetry::KeyValue::new(
-                        "service.name",
-                        provider_name,
-                    )])),
-            )
-            .install_batch(opentelemetry::runtime::Tokio)
-        {
-            Ok(t) => tracing::subscriber::set_global_default(
-                subscriber.with(tracing_opentelemetry::layer().with_tracer(t)),
-            ),
-            Err(e) => {
-                eprintln!(
-                    "Unable to configure OTEL tracing, defaulting to logging only: {:?}",
-                    e
-                );
-                tracing::subscriber::set_global_default(subscriber)
-            }
+        (Some(Ok(t)), false) => {
+            let log_layer = get_default_log_layer();
+            let tracing_layer = tracing_opentelemetry::layer().with_tracer(t);
+            let layered = base_reg.with(level_filter).with(log_layer).with(tracing_layer);
+            tracing::subscriber::set_global_default(layered)
+        }
+        (Some(Err(err)), true) => {
+            eprintln!("Unable to configure OTEL tracing, defaulting to logging only: {err:?}");
```

Is this an error if we don't supply OTEL parameters or should it just fail silently?